Source/SerializationTransformer.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  SerializationTransformer.swift
+//  Kakapo
+//
+//  Created by Alex Manzella on 27/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+/// A closure that given a String as input output the transformed String, used to transform keys of the json
+public typealias KeyTransformer = (key: String) -> String
+
+/**
+ *  A protocol that special `Serializable` objects can adopt to transform other `Serializable` objects.
+ *  At the moment only key transforms are supported, see `SnakecaseTransformer` for a concrete implementations
+ */
```

Comment:
So now we use this to only transform keys, right? For instance Serializer uses it on dictionary extension or on `serialize` method to do it. The thing is, no one prevent clients to use `keyTransformer` for other things than keys, no? So that sentence on the protocol might be a bit confusing probably.